rj_gameplay/rj_gameplay/gameplay_node.py

```diff
@@ -65,25 +81,26 @@ def __init__(self, play_selector: situation.IPlaySelector, world_state: Optional
                                                       self.create_goalie_id,
                                                       keep_latest)
 
-        self.robot_state_subs = [None] * NUM_ROBOTS
-        self.robot_intent_pubs = [None] * NUM_ROBOTS
-
-        self.override_actions = [None] * NUM_ROBOTS
+        self.override_actions: List[Optional[IAction]] = [None] * NUM_ROBOTS
 
+        # create lists for robot state subs and robot intent pubs (ROS)
+        self.robot_state_subs = []
         for i in range(NUM_ROBOTS):
-            self.robot_state_subs[i] = self.create_subscription(
+            self.robot_state_subs.append(self.create_subscription(
                 msg.RobotStatus, 'radio/robot_status/robot_' + str(i),
-                self.create_partial_robots, 10)
+                self.create_partial_robots, 10))
 
+        self.robot_intent_pubs = []
         for i in range(NUM_ROBOTS):
-            self.robot_intent_pubs[i] = self.create_publisher(
-                msg.RobotIntent, 'gameplay/robot_intent/robot_' + str(i), 10)
+            self.robot_intent_pubs.append(self.create_publisher(
+                msg.RobotIntent, 'gameplay/robot_intent/robot_' + str(i), 10))
 
         self.get_logger().info("Gameplay node started")
         self.world_state = world_state
-        self.partial_world_state: conv.PartialWorldState = None
-        self.goalie_id = None
-        self.field: rc.Field = None
+        # these 3 Nones will be filled in dynamically
+        self.partial_world_state: Optional[conv.PartialWorldState] = None
+        self.goalie_id: Optional[int] = None
+        self.field: Optional[rc.Field] = None
```

Comment:
```
Error: rj_gameplay/rj_gameplay/gameplay_node.py:84: error: Incompatible types in assignment (expression has type "None", variable has type "PartialWorldState")
Error: rj_gameplay/rj_gameplay/gameplay_node.py:86: error: Incompatible types in assignment (expression has type "None", variable has type "Field")
Error: rj_gameplay/rj_gameplay/gameplay_node.py:168: error: Argument 5 to "worldstate_creator" has incompatible type "None"; expected "int"
```

mypy doesn't like it when you initialize variables as None and fill them in dynamically, so I typed these all as Optional[<whatever type they will be>] and added a comment explaining the Nones.